mypy/checker.py

```diff
@@ -7118,6 +7118,8 @@ def flatten_types(t: Type) -> list[Type]:
     t = get_proper_type(t)
     if isinstance(t, TupleType):
         return [b for a in t.items for b in flatten_types(a)]
+    elif is_named_instance(t, "builtins.tuple"):
+        return [t.args[0]]
```

Comment:
Why is it correct to use all args in former case but just first arg in second case?

Reply:
Because `tuple` has _only_ one arg. FWIW the only real question here is whether we need to recurse, i.e. also support:
* `tuple[tuple[type[A], type[B]], ...]`
* `tuple[tuple[type[A], ...], ...]`

But TBH I don't really care, because it would be a corner case to already rare special case.

Reply:
I've decided not to, because this can be quite complex / slow with no real users asking for it.